src/c/_cffi_backend.c

```diff
@@ -619,10 +628,16 @@ get_field_name(CTypeDescrObject *ct, CFieldObject *cf);   /* forward */
 static int do_realize_lazy_struct(CTypeDescrObject *ct);
 /* forward, implemented in realize_c_type.c */
 
+static int ct_is_hidden(CTypeDescrObject *ct)
+{
+  return ((ct->ct_flags & CT_IS_OPAQUE) ||
+          (ct->ct_flags_mut & CT_UNDER_CONSTRUCTION));
```

Comment:
I'm a little skeptical of this: it feels like we haven't fully disentangled `CT_UNDER_CONSTRUCTION` from `CT_IS_OPAQUE`, but I think it's fine for now.

Reply:
I think we should entangle them fully, otherwise it will be hard to make it thread safe